client/landing/stepper/declarative-flow/start-writing.ts

```diff
@@ -0,0 +1,75 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { START_WRITING_FLOW } from '@automattic/onboarding';
+import { useSelector } from 'react-redux';
+import { recordSubmitStep } from 'calypso/landing/stepper/declarative-flow/internals/analytics/record-submit-step';
+import QuickSite from 'calypso/landing/stepper/declarative-flow/internals/steps-repository/quick-site';
+import { getCurrentUserSiteCount, isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { redirect } from './internals/steps-repository/import/util';
+import ProcessingStep from './internals/steps-repository/processing-step';
+import {
+	AssertConditionResult,
+	AssertConditionState,
+	Flow,
+	ProvidedDependencies,
+} from './internals/types';
+
+const startWriting: Flow = {
+	name: START_WRITING_FLOW,
+	useSteps() {
+		return [
+			{ slug: 'quick-site', component: QuickSite },
+			{ slug: 'processing', component: ProcessingStep },
+		];
+	},
+
+	useStepNavigation( currentStep, navigate ) {
+		const flowName = this.name;
+		function submit( providedDependencies: ProvidedDependencies = {} ) {
+			recordSubmitStep( providedDependencies, '', flowName, currentStep );
+			switch ( currentStep ) {
+				case 'quick-site':
+					return navigate( 'processing' );
+				case 'processing': {
+					if ( providedDependencies?.siteSlug ) {
+						return redirect(
+							`https://${ providedDependencies?.siteSlug }/wp-admin/post-new.php?showLaunchpad=true`
+						);
+					}
+				}
+			}
+		}
+		return { submit };
+	},
+
+	useAssertConditions(): AssertConditionResult {
+		const flowName = this.name;
+		const isLoggedIn = useSelector( isUserLoggedIn );
+		const currentUserSiteCount = useSelector( getCurrentUserSiteCount );
+		const locale = useLocale();
+
+		const logInUrl =
+			locale && locale !== 'en'
```

Comment:
Quick question: What's the difference when the locale is `en`?

Reply:
Truthfully, I implemented it this way to match the pattern in the other flows. 😅 But I believe it is to do with `/en` redirecting to the canonical url (stripping `/en` from the URL) once logged in, and then losing the values in query string. But, I haven't verified that.